rubix-common/src/main/java/com/qubole/rubix/common/utils/ClusterUtil.java

```diff
@@ -50,4 +64,66 @@ public static String getMasterHostname(Configuration conf)
     log.debug("No hostname found in etc/*-site.xml, returning localhost");
     return "localhost";
   }
+
+  public static synchronized Configuration applyRubixSiteConfig(Configuration conf) throws Exception
+  {
+    if (rubixSiteConfiguration == null) {
+      rubixSiteConfiguration  = new HashMap<>();
+      String configFileLocation = CacheConfig.getRubixSiteLocation(conf);
+      File configFile = new File(configFileLocation);
+      if (!configFile.exists()) {
+        log.warn(String.format("Config file does not exist on location: %s", configFileLocation));
```

Comment:
dont log, its okay to not have rubix-site. Just cache empty map in rubixSiteConfiguration in that case and return